File: 3931-process-string-with-special-operations-i/process-string-with-special-operations-i.py
class Solution:
    def processStr(self, s: str) -> str:
        ans = ""

        for ch in s:
            if ch == "*":
                if ans:
                    ans = ans[:-1]
            elif ch == "#":
                ans += ans
            elif ch == "%":
                ans = ans[::-1]
            else:
                ans += ch

        return ans

File: 3931-process-string-with-special-operations-i/process-string-with-special-operations-i.cs
public class Solution {
    public string ProcessStr(string s) {
        StringBuilder ans = new StringBuilder();

        foreach (char ch in s) {
            if (ch == '*') {
                if (ans.Length > 0) {
                    ans.Remove(ans.Length - 1, 1);
                }
            }
            else if (ch == '#') {
                ans.Append(ans.ToString());
            }
            else if (ch == '%') {
                var arr = ans.ToString().ToCharArray();
                Array.Reverse(arr);
                ans.Clear();
                ans.Append(arr);
            }
            else {
                ans.Append(ch);
            }
        }

        return ans.ToString();
    }
}